internal/controller/pipelinerollout_controller.go

```diff
@@ -395,19 +392,34 @@ func (r *PipelineRolloutReconciler) processExistingPipeline(ctx context.Context,
 	return nil
 }
 
-func (r *PipelineRolloutReconciler) processExistingPipelineWithoutDataLoss(ctx context.Context, pipelineRollout *apiv1.PipelineRollout, existingPipelineDef *kubernetes.GenericObject, newPipelineDef *kubernetes.GenericObject, pipelineNeedsToUpdate bool, pipelineIsUpdating bool) error {
+func (r *PipelineRolloutReconciler) processExistingPipelineWithoutDataLoss(ctx context.Context, pipelineRollout *apiv1.PipelineRollout, existingPipelineDef *kubernetes.GenericObject, newPipelineDef *kubernetes.GenericObject, pipelineNeedsToUpdate bool) error {
 
 	numaLogger := logger.FromContext(ctx)
 	var err error
 
 	var newPipelineSpec PipelineSpec
-	if err = json.Unmarshal(pipelineRollout.Spec.Pipeline.Spec.Raw, &newPipelineSpec); err != nil {
-		return fmt.Errorf("failed to convert new Pipeline spec %q into PipelineSpec type, err=%v", string(pipelineRollout.Spec.Pipeline.Spec.Raw), err)
+	if err = json.Unmarshal(newPipelineDef.Spec.Raw, &newPipelineSpec); err != nil {
+		return fmt.Errorf("failed to convert new Pipeline spec %q into PipelineSpec type, err=%v", string(newPipelineDef.Spec.Raw), err)
+	}
+	var existingPipelineSpec PipelineSpec
+	if err := json.Unmarshal(existingPipelineDef.Spec.Raw, &existingPipelineSpec); err != nil {
+		return fmt.Errorf("failed to convert existing Pipeline spec %q into PipelineSpec type, err=%v", string(existingPipelineDef.Spec.Raw), err)
 	}
 
+	// is the Pipeline currently being reconciled?
+	pipelineUpdating, err := pipelineIsUpdating(newPipelineDef, existingPipelineDef)
+	if err != nil {
+		return err
+	}
+
+	existingPipelinePauseDesired := existingPipelineSpec.Lifecycle.DesiredPhase == string(numaflowv1.PipelinePhasePaused)
```

Comment:
I needed to add this line which was not in here before, so that we only consider cases of pipeline being reconciled while it's being paused. 

Background: the reason we are checking to see if the pipeline is being reconciled still is to make sure that we allow the pipeline update to go completely all the way through before we run it again. We don't want to resume it until this has completed.

But this makes it so we only concern ourselves of this while we have actually paused the pipeline.

Otherwise, we incorrectly will pause the pipeline simply because we are in the middle of reconciling the change to set `desiredPhase=Running`